libchisel/src/gc.rs

```diff
@@ -0,0 +1,119 @@
+                update_section!(module.table_section_mut(), gced.table_section(), TableSection::with_entries(vec![]));
+                update_section!(module.memory_section_mut(), gced.memory_section(), MemorySection::with_entries(vec![]));
+                update_section!(module.global_section_mut(), gced.global_section(), GlobalSection::with_entries(vec![]));
+                update_section!(module.export_section_mut(), gced.export_section(), ExportSection::with_entries(vec![]));
+
+                // Awaiting mainlining of https://github.com/paritytech/parity-wasm/pull/250
+                // match (module.start_section_mut(), gced.start_section()) {
+                //     (Some(ssm), Some(ss)) => *ssm = ss.clone(),
+                //     (Some(ssm), None) => *ssm = StartSection::with_entries(vec![]),
+                //     _ => {}
+                // }
+
+                update_section!(module.elements_section_mut(), gced.elements_section(), ElementSection::with_entries(vec![]));
+                update_section!(module.code_section_mut(), gced.code_section(), CodeSection::with_bodies(vec![]));
+                update_section!(module.data_section_mut(), gced.data_section(), DataSection::with_entries(vec![]));
+
+                Ok(gced_bytes.len() != serialized.len())
+            },
+            Err(e) => Err(format!("GC failure: {}", e))
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use parity_wasm::{deserialize_buffer};
+
+    #[test]
+    fn do_not_touch_simple_module() {
+        let wasm: Vec<u8> = vec![0x00, 0x61, 0x73, 0x6d, 0x01, 0x00, 0x00, 0x00];
+
+        let mut module = deserialize_buffer::<Module>(&wasm).unwrap();
+        let result = WasmGC::default().translate(&mut module).unwrap();
+        assert_eq!(false, result);
+    }
+
+    #[test]
+    fn gc_does_something() {
+        let wasm: Vec<u8> = vec![0, 97, 115, 109, 1, 0, 0, 0, 1, 18, 4, 96, 1, 127, 1, 127, 96, 0, 1, 127, 96, 0, 0, 96, 1, 127, 1, 127, 2, 77, 5, 3, 101, 110, 118, 10, 109, 101, 109, 111, 114, 121, 66, 97, 115, 101, 3,127, 0, 3, 101, 110, 118, 6, 109, 101, 109, 111, 114, 121, 2, 0, 128, 2, 3, 101, 110, 118, 5, 116, 97, 98, 108, 101, 1, 112, 0, 0, 3, 101, 110, 118, 9, 116, 97, 98, 108, 101, 66, 97, 115, 101, 3, 127, 0, 3, 101, 110, 118, 5, 95, 112, 117, 116, 115, 0, 3, 3, 4, 3, 1, 2, 2, 6, 16, 3, 127, 1, 65, 0, 11, 127, 1, 65, 0, 11, 127, 0, 65, 0, 11, 7, 51, 4, 18, 95, 95, 112, 111, 115, 116,95, 105, 110, 115, 116, 97, 110, 116, 105, 97, 116, 101, 0, 3, 5, 95, 109, 97, 105, 110, 0, 1, 11, 114, 117, 110, 80, 111, 115, 116, 83, 101, 116, 115, 0, 2, 4, 95, 115, 116, 114, 3, 4, 9, 1, 0, 10, 37, 3, 9, 0, 35, 0, 16, 0, 26, 65, 0, 11, 3, 0, 1, 11, 21, 0, 35, 0, 65, 16, 106, 36, 2, 35, 2, 65, 128, 128, 192, 2, 106, 36, 3, 16, 2, 11, 11, 19, 1, 0, 35, 0, 11, 13, 104, 101, 108, 108, 111, 44, 32, 119, 111, 114, 108, 100, 33];
```

Comment:
Can all these just use `from_hex`? Easier to copy paste from files.

Reply:
what makes it easier to read from file? I have my scripts that create those lists automatically.

Reply:
With `from_hex` there's no need for a script. However this is not crticial for this PR.